text/docs-code-contributions.md

```diff
@@ -0,0 +1,59 @@
+- Start Date: 2019-02-07)
+- RFC PR: 
+- Gatsby Issue:
+
+# Summary
+
+Reorganize the contributor documentation to aid in Gatsby core and docs development.
+
+# Motivation
+
+The Gatsby.js docs, while wonderful, are light on coding instructions for contributors. With dedicated sections in the Contributing docs, it will be easier to learn how to clone the repository, where to find certain things like docs or plugins, how to run tests, and insider infrastructure knowledge required to submit a successful pull request on the first try.
+
+A lot of this information already lives on the "How to Contribute" page, but it's buried. It would help discoverability if this information lived on separate pages.
+
+# Detailed design
+
+I propose that we add some pages to the existing contributor documentation and move some content around. Here is an IA diagram that (briefly) reflects the current structure and what I think we should add (open for discussion, of course):
+
+Contributing
+- Community
+    - Pair Programming Sessions
+    - Where to participate in the community (this was in RFCs, perhaps it could be reintroduced and content moved from "How to contribute"?)
+    - How to run a Gatsby workshop*
+    - How to pitch Gatsby*
+- Code of Conduct
+- Gatsby Style Guide
+- How to contribute
```

Comment:
In this section, I wonder if it would make sense to group by major tasks? For example:
- Managing issues (e.g. labeling, giving feedback)
- Reviewing pull requests
- Docs contributions
- Code contributions
- Community contributions (e.g. blog, events)
- Showcase contributions

I think the thing that trips me up here is the free-form-ish nature of the bullets. If we could get things grouped by the major questions, e.g. "Why", "What", "Where", "How", it makes this a little more sequential, and people can lead themselves deeper into contributing.

(Does that make sense to anyone who's not inside my head right now?)

We could add an additional top-level section for "Why contribute to Gatsby?" that goes into all the amazing things that our community represents, as well as the perks we offer to contributors.

Reply:
On the free-form-ish-nature of my PR: something we should do is come up with a syntax for indicating page structure while providing comments/notations. That's probably what threw you off 🙈

I'm totally on board with grouping by major tasks. Something that I'm also considering, though, is the persona of each one, such as filing issues vs. managing issues. The current content structure leans more heavily towards beginner contributions (i.e. how to file an issue vs. Managing issues).

Perhaps we could group by tasks but also introduce some pages to cover more than one persona? (e.g. the beginner and the regular/more advanced contributor)

Such as:
- How to file an issue
- Managing issues & pull requests
- Setting up your local dev environment (Mike's suggestion, this could cover multiple personas)
- Docs contributions
- Code contributions

Reply:
Really like the structure you proposed in the comment seems great @marcysutton!

Reply:
I'm into this for sure. I think persona-based with task grouping makes a lot of sense.

Reply:
@jlengstorf idea from last year: to create a gatsbyjs.org/developers page that has its own sidebar, so all these new items would have room to breath and contributing docs wouldn't be at the bottom of a list anymore!

I like these ideas. "-ing" verbs that focus on tasks, and organizing by persona makes sense.

Reply:
Thanks for following up @shannonbux! It's definitely good to consider right now whether we want this new section to be called "developers" or "contributing". I'm leaning towards the latter so that it is encouraging to people who aren't developers...that seems consistent with the Gatsby story. Any strong opinions on this, team?

Reply:
I like the idea of "contributing". In addition to the reason you mentioned, it's more straightforward; most SaaS tools are for non-developers so "developers" is immediately clear that it is referring to API documentation. Because Gatsby is for developers, "contributing" is more straightforward as well as encouraging/aspirational!